beacon-chain/sync/rpc_send_request.go

```diff
@@ -169,9 +169,9 @@ func SendBlobsByRangeRequest(ctx context.Context, tor blockchain.TemporalOracle,
 	}
 	defer closeStream(stream, log)
 
-	maxBlobsPerBlock := uint64(params.BeaconConfig().MaxBlobsPerBlock(req.StartSlot))
+	maxBlobsPerBlock := uint64(params.BeaconConfig().MaxBlobsPerBlock(req.StartSlot + primitives.Slot(req.Count)))
```

Comment:
This needs to be addressed with a unit test. Do not merge a fix here without unit tests. 

This fix shouldn't be part of this PR anyway, imo. Your PR should be only e2e changes.